src/test/resources/org/csanchez/jenkins/plugins/kubernetes/pipeline/declarativePodWithInheritFrom.groovy

```diff
@@ -0,0 +1,23 @@
+pipeline {
+  agent {
+    kubernetes {
+      label 'mypod'
+      podTemplate {
+        inheritFrom 'maven'
```

Comment:
+1: Having `inheritFrom` like this (under podTemplate) makes much more sense!

Reply:
So I guess we need to decide should there be a `label` under `agent` and `kubernetes` and if so why and do they have different semantics?

Also should there be `inheritFrom` under `kubernetes` and `podTemplate`.

FWIW my 2 main use cases are:

1. refer to a podTemplate by name/label
2. inheritFrom a named podTemplate and change something (e.g. the maven containerTemplate's image version)

I guess for (1) we can just use:
```
agent {
  label "nameOfPodTemplate"
}
```

then for (2) we maybe go with something like this?
```
agent {
  kubernetes {
    podTemplate {
       inheritFrom 'maven'
      containerTemplate { 
          name  "maven"
          image "mycustom-maven-image"
```

Reply:
+1 I totally agree on how these two use cases are handled.

Please note that on (1) its labelOfPodTemplate and not the name.

Reply:
I guess its clear there's some confusion though over the current properties; so might be worth a bit of review and maybe some deprecation?

Reply:
I guess its clear that the use of `label` just refers to an existing podTemplate without any modification and `inheritFrom` always creates a new podTemplate by combining inline customisation with an existing podTemplate. Its just a tad confusing that there are so many places you can use `label` and 'inheritFrom`.

Maybe we deprecate `kubernetes.label` and `kubernetes.inheritFrom`? Then you either use `agent.label` to use a pod template via label or `agent.kubernetes.podTemplate.inheritFrom` to create a new PodTemplate (and maybe override some configurations)?

Reply:
+1000

Reply:
So I can't remember why `kubernetes.label` exists. It's been a while. =) But `agent { label 'foo' }` will just try to run on an agent with the label `foo`, regardless of how that agent is created/provisioned/etc. So if you have the Kubernetes cloud set up on the Jenkins master with a pod template defined with the label `foo`, an agent will get spun up and used for that Pipeline. But if you have, say, an SSH agent set up with the label `foo` and it's got a free executor, that'll get used before it tries to provision any new agents. 

And yeah, if the current functionality of `kubernetes { label ... }` is just to define a label on the `podTemplate` that gets created behind the scenes, I'd deprecate it in favor of the better `podTemplate` configuration options.